lib/puppet/functions/op/hiera.rb

```diff
@@ -180,12 +186,77 @@ def lookup_key(secret_name, options, context)
 
     # not found
     if secretvalue.nil?
-      context.explain { "OP: Unable to find secret #{secret_title} in vault #{vault}#{msg}" }
-      context.not_found
+      if create
```

Comment:
For long blocks, I think Ruby recommends using return to eliminate the conditions you don't care, then you can have the code enclosed in your conditions to the same level as normal code for easy reading, for example for this line, you could do:

```
return unless create
..... (this could be separated to its own function)
```